client/client.go

```diff
@@ -542,10 +586,16 @@ func (c *client) GetLocalTSAsync(ctx context.Context, dcLocation string) TSFutur
 	req.keyspaceID = c.keyspaceID
 	req.dcLocation = dcLocation
 
-	if err := c.tsoClient.dispatchRequest(dcLocation, req); err != nil {
+	tsoClient := c.getTSOClient()
+	if tsoClient == nil {
+		req.done <- errs.ErrClientGetTSO
+		return req
+	}
```

Comment:
Maybe we can do this check ASAP right after `req` is constructed to prevent some unnecessary variable copying and assignment.

Reply:
You're right. Moved.

```
	req := tsoReqPool.Get().(*tsoRequest)
	tsoClient := c.getTSOClient()
	if tsoClient == nil {
		req.done <- errs.ErrClientGetTSO
		return req
	}

	req.requestCtx = ctx
	req.clientCtx = c.ctx
	req.start = time.Now()
	req.keyspaceID = c.keyspaceID
	req.dcLocation = dcLocation
```

Reply:
I reverted, because the test failures show that after "req.done <- errs.ErrClientGetTSO", the TSFuture.Wait() could use some fields whose initialization are skipped here.